src/snowflake/connector/bind_upload_agent.py

```diff
@@ -38,13 +39,29 @@ def __init__(
             rows: Rows of binding parameters in CSV format.
             stream_buffer_size: Size of each file, default to 10MB.
         """
+        self._use_scoped_temp_object = (
+            cursor.connection._session_parameters.get(
+                _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING, False
+            )
+            if cursor.connection._session_parameters
+            else False
+        )
         self.cursor = cursor
+        self._stage_name = (
+            random_name_for_temp_object(TempObjectType.STAGE)
+            if self._use_scoped_temp_object
+            else "SYSTEMBIND"
+        )
```

Comment:
actually I just realized that if `_use_scoped_temp_object` is True, then each time when we instantiate a BindUploadAgent (we do this when binding parameters), a new temp stage created.

is this the same behavior as the sproc connector change?

Reply:
This is a tricky problem, the original design is that the temp stage name in bind upload agent is a constant. So that there is only one temp stage created by it.
what i can think of is creating a constant stage name following the naming rule, what do you think?